test/test_eos_unit.cpp

```diff
@@ -475,9 +471,8 @@ SCENARIO("Ideal gas entropy", "[IdealGas][Entropy]") {
     constexpr Real EntropyT0 = 100;
     constexpr Real EntropyRho0 = 1e-03;
     EOS host_eos = IdealGas(gm1, Cv, EntropyT0, EntropyRho0);
-    EOS eos = host_eos.GetOnDevice();
```

Comment:
Why is there an issue with this ideal gas entropy test but not the vector ideal gas tests below? I would advocate that we fix this test on device rather than preventing it from running on device.

Reply:
There's nothing wrong with this test, but it wasn't being run on device. The call was being made on the scalar quantities on a "device" object in a host execution context. It only worked because ideal gas is the same object on host and device. But it's technically "paradigm" breaking. One should either put the call in a `par_for` or remove the `GetOnDevice()`. I chose the former since it was a smaller change and the vector calls below test the device context. We can go the other route if you'd prefer though.

Reply:
Ah thank you. That makes sense. I'm glad now that I included the entropy in the vector test then.